### PR TITLE
Document recently added utility compiler passes.

### PR DESCRIPTION
**Pull request by ScottTodd**
Following up on
* https://github.com/openxla/iree/pull/16550#discussion_r1501003411
* https://github.com/openxla/iree/pull/16970#discussion_r1552094712

### CODE REVIEW

File: docs/website/docs/developers/general/developer-tips.md [hunk 1]
```diff
@@ -381,6 +381,38 @@ Flag | Files dumped
...
+$ iree-benchmark-module \
+  /tmp/iree/simple_abs/module_abs_dispatch_0_benchmark.vmfb
+```
+
+The _binary_ files produced by `--iree-hal-dump-executable-binaries-to`
+can be passed to `iree-benchmark-executable` where they are benchmarked
+directly, without using the IREE VM, HAL APIs, task system, etc. Note that this
+interface is much lower level.
+
+<!-- TODO: example? -->
```
Comment (ScottTodd) [inline docs/website/docs/developers/general/developer-tips.md]: I tried this, is the output as expected? Not sure what the text at the start is 🤔 

```
D:\dev\projects\iree (docs-tips-new-passes)
λ ..\iree-build\tools\iree-compile.exe \
  .\samples\models\simple_abs.mlir \
  --iree-hal-dump-executable-binaries-to=D:\dev\projects\iree-tmp\simple_abs_binaries \
  --iree-hal-target-backends=llvm-cpu \
  -o NUL

D:\dev\projects\iree (docs-tips-new-passes)
λ ..\iree-build\tools\iree-benchmark-executable.exe \
  --device=local-sync \
  --executable_format=embedded-elf-x86_64 \
  --executable_file=D:\dev\projects\iree-tmp\simple_abs_binaries\module_abs_dispatch_0_embedded_elf_x86_64.so \
  --entry_point=0 \
  --binding=f32=2.5 \
  --binding=f32=0 \
  --workgroup_count=1,1,1

===----------------------------------------------------------------------===

         ΓûêΓûê     ΓûêΓûê  ΓûêΓûêΓûêΓûêΓûê  ΓûêΓûêΓûêΓûêΓûêΓûê  ΓûêΓûêΓûê    ΓûêΓûê ΓûêΓûê ΓûêΓûêΓûê    ΓûêΓûê  ΓûêΓûêΓûêΓûêΓûêΓûê
         ΓûêΓûê     ΓûêΓûê ΓûêΓûê   ΓûêΓûê ΓûêΓûê   ΓûêΓûê ΓûêΓûêΓûêΓûê   ΓûêΓûê ΓûêΓûê ΓûêΓûêΓûêΓûê
   ΓûêΓûê ΓûêΓûê
         ΓûêΓûê  Γûê  ΓûêΓûê ΓûêΓûêΓûêΓûêΓûêΓûêΓûê ΓûêΓûêΓûêΓûêΓûêΓûê  ΓûêΓûê ΓûêΓûê  ΓûêΓûê ΓûêΓûê ΓûêΓûê ΓûêΓûê  ΓûêΓûê ΓûêΓûê   ΓûêΓûêΓûê
         ΓûêΓûê ΓûêΓûêΓûê ΓûêΓûê ΓûêΓûê   ΓûêΓûê ΓûêΓûê   ΓûêΓûê ΓûêΓûê  ΓûêΓûê ΓûêΓûê ΓûêΓûê ΓûêΓûê
  ΓûêΓûê ΓûêΓûê ΓûêΓûê    ΓûêΓûê
          ΓûêΓûêΓûê ΓûêΓûêΓûê  ΓûêΓûê   ΓûêΓûê ΓûêΓûê   ΓûêΓûê ΓûêΓûê   ΓûêΓûêΓûêΓûê ΓûêΓûê ΓûêΓûê
 ΓûêΓûêΓûêΓûê  ΓûêΓûêΓûêΓûêΓûêΓûê

===----------------------------------------------------------------------===

Tracing is enabled and will skew your results!
The timings involved here can an order of magnitude off due to the tracing
time sampling, recording, and instrumentation overhead. Disable tracing with
IREE_ENABLE_RUNTIME_TRACING=OFF and rebuild.

2024-04-08T09:40:10-07:00
Running ..\iree-build\tools\iree-benchmark-executable.exe
Run on (64 X 3693 MHz CPU s)
CPU Caches:
  L1 Data 32 KiB (x32)
  L1 Instruction 32 KiB (x32)
  L2 Unified 512 KiB (x32)
  L3 Unified 16384 KiB (x8)
***WARNING*** Library was built as DEBUG. Timings may be affected.
---------------------------------------------------------------------------------------------------
Benchmark                                         Time             CPU   Iterations UserCounters...
---------------------------------------------------------------------------------------------------
BM_dispatch_1x1x1/process_time/real_time       5466 ns         7849 ns       133376 items_per_second=182.959k/s
```

Comment (benvanik) [reply docs/website/docs/developers/general/developer-tips.md]: probably? (not sure I understand the question?)

File: docs/website/docs/developers/general/developer-tips.md [hunk 2]
```diff
@@ -381,6 +381,38 @@ Flag | Files dumped
...
+
+$ iree-compile \
+  /tmp/iree/simple_abs/module_abs_dispatch_0_embedded_elf_x86_64_benchmark.mlir \
+  -o /tmp/iree/simple_abs/module_abs_dispatch_0_benchmark.vmfb
+
+$ iree-benchmark-module \
+  /tmp/iree/simple_abs/module_abs_dispatch_0_benchmark.vmfb
+```
+
+The _binary_ files produced by `--iree-hal-dump-executable-binaries-to`
```
Comment (benvanik) [inline docs/website/docs/developers/general/developer-tips.md:406]: probably worth nesting in a low-level dispatch benchmarking or something, given that it's a different flow than above

**Review by benvanik — APPROVED** [review]

File: docs/website/docs/developers/general/developer-tips.md (hunk 1 above)
Comment (ScottTodd) [reply docs/website/docs/developers/general/developer-tips.md]: The output at the bottom looks normal, but idk what this is:
```
===----------------------------------------------------------------------===

         ΓûêΓûê     ΓûêΓûê  ΓûêΓûêΓûêΓûêΓûê  ΓûêΓûêΓûêΓûêΓûêΓûê  ΓûêΓûêΓûê    ΓûêΓûê ΓûêΓûê ΓûêΓûêΓûê    ΓûêΓûê  ΓûêΓûêΓûêΓûêΓûêΓûê
         ΓûêΓûê     ΓûêΓûê ΓûêΓûê   ΓûêΓûê ΓûêΓûê   ΓûêΓûê ΓûêΓûêΓûêΓûê   ΓûêΓûê ΓûêΓûê ΓûêΓûêΓûêΓûê
   ΓûêΓûê ΓûêΓûê
         ΓûêΓûê  Γûê  ΓûêΓûê ΓûêΓûêΓûêΓûêΓûêΓûêΓûê ΓûêΓûêΓûêΓûêΓûêΓûê  ΓûêΓûê ΓûêΓûê  ΓûêΓûê ΓûêΓûê ΓûêΓûê ΓûêΓûê  ΓûêΓûê ΓûêΓûê   ΓûêΓûêΓûê
         ΓûêΓûê ΓûêΓûêΓûê ΓûêΓûê ΓûêΓûê   ΓûêΓûê ΓûêΓûê   ΓûêΓûê ΓûêΓûê  ΓûêΓûê ΓûêΓûê ΓûêΓûê ΓûêΓûê
  ΓûêΓûê ΓûêΓûê ΓûêΓûê    ΓûêΓûê
          ΓûêΓûêΓûê ΓûêΓûêΓûê  ΓûêΓûê   ΓûêΓûê ΓûêΓûê   ΓûêΓûê ΓûêΓûê   ΓûêΓûêΓûêΓûê ΓûêΓûê ΓûêΓûê
 ΓûêΓûêΓûêΓûê  ΓûêΓûêΓûêΓûêΓûêΓûê

===----------------------------------------------------------------------===
```

Comment (ScottTodd) [reply docs/website/docs/developers/general/developer-tips.md]: Ah. I can run https://github.com/openxla/iree/blob/main/tools/test/iree-benchmark-executable.mlir to compare the output.

Comment (ScottTodd) [reply docs/website/docs/developers/general/developer-tips.md]: I see the same thing with that file. Going to ignore it :P

File: docs/website/docs/developers/general/developer-tips.md (hunk 2 above)
Comment (ScottTodd) [reply docs/website/docs/developers/general/developer-tips.md:406]: Good idea, done.

File: docs/website/docs/developers/general/developer-tips.md (hunk 1 above)
Comment (benvanik) [reply docs/website/docs/developers/general/developer-tips.md]: https://github.com/openxla/iree/blob/336ba1278c410b7dd46c2d01acbaa8ab29fc7b9a/runtime/src/iree/testing/benchmark_full.cc#L166-L189

Comment (ScottTodd) [reply docs/website/docs/developers/general/developer-tips.md]: Oh! 😂

Comment (benvanik) [reply docs/website/docs/developers/general/developer-tips.md]: `The timings involved here can an order `
argghhhhh

**Pull request merged**